File: classes/parte-3-valores-tipos-y-variables/053-nulabilidad-null-nil-none-option-y-valores-ausentes/implementaciones/c/main.c
#include <stdio.h>

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    if (n == 0) {
        printf("valor=ausente\n");
    } else {
        printf("valor=%ld\n", n);
    }
    return 0;
}

File: classes/parte-3-valores-tipos-y-variables/053-nulabilidad-null-nil-none-option-y-valores-ausentes/implementaciones/rust/main.rs
use std::io::Read;

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let n: i64 = s.trim().parse().unwrap();
    let valor: Option<i64> = if n == 0 { None } else { Some(n) };
    match valor {
        None => println!("valor=ausente"),
        Some(v) => println!("valor={v}"),
    }
}
